src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawWriter.cs

```diff
@@ -185,7 +185,74 @@ public override void WriteValue(object value)
         {
             ArgumentNullException.ThrowIfNull(value);
 
-            WriteString(XmlUntypedConverter.Untyped.ToString(value, _resolver));
+            Type sourceType = value.GetType();
+            if (!TypeScope.IsKnownType(sourceType))
+            {
+                WriteString(XmlUntypedConverter.Untyped.ToString(value, _resolver));
+                return;
+            }
+
+#pragma warning disable IL2026
```

Comment:
This is something new for me. Should I add some warning?
Am I allowed to use it at all ?

Reply:
Fixed by using TypeCode